### DISCUSSION TITLE
Invalid register: v17. Must be between v0 and v15, inclusive.

### DISCUSSION THREAD

**Discussion by ghost** [iBotPeaches/Apktool#3666] [Q&A]
I encountered an APK that is using `.locals 17` under OnCreate function in `com/ctugames/km2/TiredBirds.smali `

`invoke-static {p0}, Lcom/xxx/xxx/Main;->Start(Landroid/content/Context;)V` 

like

```
.method public final onCreate(Landroid/os/Bundle;)V
    .locals 17

    invoke-static {p0}, Lcom/xxx/xxx/Main;->Start(Landroid/content/Context;)V
```

But Apktool does not let me compile

```
I: Using Apktool 2.9.3
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
com.ctugames.km2\smali\com\ctugames\km2\TiredBirds.smali[1951,4] Invalid register: v17. Must be between v0 and v15, inclusive.
Could not smali file: com/ctugames/km2/TiredBirds.smali
```

I changed it to `.locals 16` which is max register, but it crashes

I never had this problem adding a line below OnCreate until now. What should I do?

APK: https://apkcombo.com/the-wild-darkness/com.ctugames.km2/

**Comment by iBotPeaches** [comment]
Well thats odd - if Apktool (smali) disassembled into an invalid state - either it was intentionally like that to thwart this type of effort or a bug in smali.

**Comment by ghost** [comment]
Nevermind, I replaced `invoke-static {p0}` with `invoke-static/range {p0 .. p0}` and it works now
